### PR TITLE
Dumb component login DESKTOP-840

### PR DESCRIPTION
**Pull request by awendland**
Added dumb components for shared/login/intro and for shared/login/login.

The file/component names in shared/login is a tad confusing, but I tried to guess at the use cases by looking at import statements and walking through the app registration process. I've renamed "UsernameEmail" to "UsernameEmail (Login)" to reflect that it seems to only be used in the login process (when the app is first installed), not in the signup process as denoted by the folder it's located in and the method it's loaded by (some injection through actions/dispatch).

@keybase/react-hackers

### CODE REVIEW

**Comment by awendland** [comment]
#### Intro
screen shot 2016-06-02 at 4 26 20 pm
screen shot 2016-06-02 at 4 26 26 pm


#### Login
screen shot 2016-06-02 at 4 25 56 pm
screen shot 2016-06-02 at 4 27 40 pm

**Comment by keybase-ci-visdiff** [comment]
:mag_right: These commits introduced some visual changes:
 * Added: **Username_Email_Login-Waiting**  
   Username_Email_Login-Waiting rendered
 * Added: **Username_Email_Login-Start**  
   Username_Email_Login-Start rendered
 * Added: **Username_Email_Login-Name_Error**  
   Username_Email_Login-Name_Error rendered
 * Added: **Username_Email_Login-Name_Email**  
   Username_Email_Login-Name_Email rendered
 * Added: **Username_Email_Login-Name**  
   Username_Email_Login-Name rendered
 * Added: **Username_Email_Login-Email_Error**  
   Username_Email_Login-Email_Error rendered
 * Added: **Username_Email_Login-Email**  
   Username_Email_Login-Email rendered
 * Added: **Login_Signed_Out-Single_previous_user** [(view)](http://keybase-app-visdiff.s3.amazonaws.com/1464911861107-92d2850f515b-30bc9854894c/Login_Signed_Out-Single_previous_user-30bc9854894c.png)
 * Added: **Login_Signed_Out-Multiple_previous_users** [(view)](http://keybase-app-visdiff.s3.amazonaws.com/1464911861107-92d2850f515b-30bc9854894c/Login_Signed_Out-Multiple_previous_users-30bc9854894c.png)
 * Added: **Login_Signed_Out-Error** [(view)](http://keybase-app-visdiff.s3.amazonaws.com/1464911861107-92d2850f515b-30bc9854894c/Login_Signed_Out-Error-30bc9854894c.png)
 * Added: **Intro-User_who_just_revoked_device** [(view)](http://keybase-app-visdiff.s3.amazonaws.com/1464911861107-92d2850f515b-30bc9854894c/Intro-User_who_just_revoked_device-30bc9854894c.png)
 * Added: **Intro-First_time_user** [(view)](http://keybase-app-visdiff.s3.amazonaws.com/1464911861107-92d2850f515b-30bc9854894c/Intro-First_time_user-30bc9854894c.png)
 * Removed: **Username_Email-Waiting** [(view original)](http://keybase-app-visdiff.s3.amazonaws.com/1464911861107-92d2850f515b-30bc9854894c/Username_Email-Waiting-92d2850f515b.png)
 * Removed: **Username_Email-Start** [(view original)](http://keybase-app-visdiff.s3.amazonaws.com/1464911861107-92d2850f515b-30bc9854894c/Username_Email-Start-92d2850f515b.png)
 * Removed: **Username_Email-Name_Error** [(view original)](http://keybase-app-visdiff.s3.amazonaws.com/1464911861107-92d2850f515b-30bc9854894c/Username_Email-Name_Error-92d2850f515b.png)
 * Removed: **Username_Email-Name_Email** [(view original)](http://keybase-app-visdiff.s3.amazonaws.com/1464911861107-92d2850f515b-30bc9854894c/Username_Email-Name_Email-92d2850f515b.png)
 * Removed: **Username_Email-Name** [(view original)](http://keybase-app-visdiff.s3.amazonaws.com/1464911861107-92d2850f515b-30bc9854894c/Username_Email-Name-92d2850f515b.png)
 * Removed: **Username_Email-Email_Error** [(view original)](http://keybase-app-visdiff.s3.amazonaws.com/1464911861107-92d2850f515b-30bc9854894c/Username_Email-Email_Error-92d2850f515b.png)
 * Removed: **Username_Email-Email** [(view original)](http://keybase-app-visdiff.s3.amazonaws.com/1464911861107-92d2850f515b-30bc9854894c/Username_Email-Email-92d2850f515b.png)

File: shared/login/forms/dumb.desktop.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  onSignup: () => {},
+  onLogin: () => {},
+  justRevokedSelf: null
+}
+
+export const dumbMap: DumbComponentMap<Intro> = {
+  component: Intro,
+  mocks: {
+    'First time user': props,
+    'User who just revoked device': Object.assign({}, props, {
```
Comment (chromakode) [inline shared/login/forms/dumb.desktop.js]: I believe we in general prefer the `...props` style to `Object.assign`

File: shared/login/login/dumb.desktop.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  showTypingChange: createLogger('showTypingChange'),
+  saveInKeychainChange: createLogger('saveInKeychainChange'),
+  onSubmit: createLogger('onSubmit')
+}
+
+export const dumbMap: DumbComponentMap<Login> = {
+  component: Login,
+  mocks: {
+    'Single previous user': props,
+    'Error': Object.assign({}, props, {
```
Comment (chromakode) [inline shared/login/login/dumb.desktop.js]: Ditto here re `...props` style

**Comment by chromakode** [comment]
:+1: % style nits

**Comment by keybase-ci-visdiff** [comment]
:mag_right: These commits introduced some visual changes:
 * Added: **Username_Email_Login-Waiting**  
   Username_Email_Login-Waiting rendered
 * Added: **Username_Email_Login-Start**  
   Username_Email_Login-Start rendered
 * Added: **Username_Email_Login-Name_Error**  
   Username_Email_Login-Name_Error rendered
 * Added: **Username_Email_Login-Name_Email**  
   Username_Email_Login-Name_Email rendered
 * Added: **Username_Email_Login-Name**  
   Username_Email_Login-Name rendered
 * Added: **Username_Email_Login-Email_Error**  
   Username_Email_Login-Email_Error rendered
 * Added: **Username_Email_Login-Email**  
   Username_Email_Login-Email rendered
 * Added: **Login_Signed_Out-Single_previous_user** [(view)](http://keybase-app-visdiff.s3.amazonaws.com/1464986068574-be221f408f4e-b0f221a97018/Login_Signed_Out-Single_previous_user-b0f221a97018.png)
 * Added: **Login_Signed_Out-Multiple_previous_users** [(view)](http://keybase-app-visdiff.s3.amazonaws.com/1464986068574-be221f408f4e-b0f221a97018/Login_Signed_Out-Multiple_previous_users-b0f221a97018.png)
 * Added: **Login_Signed_Out-Error** [(view)](http://keybase-app-visdiff.s3.amazonaws.com/1464986068574-be221f408f4e-b0f221a97018/Login_Signed_Out-Error-b0f221a97018.png)
 * Added: **Intro-User_who_just_revoked_device** [(view)](http://keybase-app-visdiff.s3.amazonaws.com/1464986068574-be221f408f4e-b0f221a97018/Intro-User_who_just_revoked_device-b0f221a97018.png)
 * Added: **Intro-First_time_user** [(view)](http://keybase-app-visdiff.s3.amazonaws.com/1464986068574-be221f408f4e-b0f221a97018/Intro-First_time_user-b0f221a97018.png)
 * Removed: **Username_Email-Waiting** [(view original)](http://keybase-app-visdiff.s3.amazonaws.com/1464986068574-be221f408f4e-b0f221a97018/Username_Email-Waiting-be221f408f4e.png)
 * Removed: **Username_Email-Start** [(view original)](http://keybase-app-visdiff.s3.amazonaws.com/1464986068574-be221f408f4e-b0f221a97018/Username_Email-Start-be221f408f4e.png)
 * Removed: **Username_Email-Name_Error** [(view original)](http://keybase-app-visdiff.s3.amazonaws.com/1464986068574-be221f408f4e-b0f221a97018/Username_Email-Name_Error-be221f408f4e.png)
 * Removed: **Username_Email-Name_Email** [(view original)](http://keybase-app-visdiff.s3.amazonaws.com/1464986068574-be221f408f4e-b0f221a97018/Username_Email-Name_Email-be221f408f4e.png)
 * Removed: **Username_Email-Name** [(view original)](http://keybase-app-visdiff.s3.amazonaws.com/1464986068574-be221f408f4e-b0f221a97018/Username_Email-Name-be221f408f4e.png)
 * Removed: **Username_Email-Email_Error** [(view original)](http://keybase-app-visdiff.s3.amazonaws.com/1464986068574-be221f408f4e-b0f221a97018/Username_Email-Email_Error-be221f408f4e.png)
 * Removed: **Username_Email-Email** [(view original)](http://keybase-app-visdiff.s3.amazonaws.com/1464986068574-be221f408f4e-b0f221a97018/Username_Email-Email-be221f408f4e.png)

**Pull request merged**